src/Shared/Microsoft.DotNet.Kusto/SharedKustoQueries.cs

```diff
@@ -0,0 +1,89 @@
+using System;
+using System.Data;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.DotNet.Kusto
+{
+    public static class SharedKustoQueries
+    {
+        public static MultiProjectKustoQuery CreateBuildTimesQueries(string repository, string branch, int days)
```

Comment:
It's a bit weird for this to be in the "Kusto" DLL. This is going to get shared with other repositories, like Helix, and helix shouldn't be making weird queries to the Timeline tables... I don't think?  This is pretty Maestro specific stuff.

Reply:
Really it's a bit weird to share _the queries_.  It would make more sense to expose some TimelineDataProvider, share that, and it would have these queries inside it.  Sharing the raw queries is sort of like sharing the body of a method... without something to interpret the results, there is a really weird binding between the query and the interpreter.

---
In reply to: [384082065](https://github.com/dotnet/arcade-services/pull/990#discussion_r384082065) [](ancestors = 384082065)

Reply:
I wasn't sure where to put it. I want it to be shared between Maestro.Web and DependencyUpdater. Maybe another shared Maestro project?

I'll come talk to you so I understand your idea of the TimelineDataProvider.